src/hooks/useSubnameAbilities.ts

```diff
@@ -0,0 +1,64 @@
+import { useMemo } from 'react'
+import { useAccount } from 'wagmi'
+
+import { useGetWrapperData } from './useGetWrapperData'
+import { useNameDetails } from './useNameDetails'
+import { useSelfAbilities } from './useSelfAbilities'
+
+type OwnerData = ReturnType<typeof useNameDetails>['ownerData']
+
+type ReturnData = {
+  canDelete: boolean
+  canDeleteContract?: 'registry' | 'nameWrapper'
+}
+
+export const useSubnameAbilities = (name: string, ownerData: OwnerData): ReturnData => {
+  const nameHasOwner = !!ownerData
+  const nameParts = name?.split('.') || []
+  const isValid2LD = nameParts.length === 3 && nameParts[2] === 'eth'
```

Comment:
For deleting subnames, we are limiting it to 2LD for now since we don't want to recursively delete all the sub subnames for the beta launch

Reply:
Okay, let's get rid of this in favour of checking children of a subdomain. We may want to add some kind of tooltip to show it can't be deleted because it has children.